lib/tasks/dev.rake

```diff
@@ -184,13 +184,12 @@ namespace :openfoodnetwork do
 
       enterprise2 = Enterprise.find_by_name('Enterprise 2')
       enterprise2.sells = 'any'
-      enterprise2.shipping_methods.build(
-        name: 'Pickup',
-        zone_id: 3,
-        require_ship_address: true,
-        calculator_type: 'OpenFoodNetwork::Calculator::Weight',
-        distributor_ids: [enterprise2.id]
-      )
+      enterprise2.shipping_methods << FactoryBot.create(:shipping_method,
```

Comment:
I always saw FactoryBot's purpose to support specs. It would be better to avoid it in other parts of the code to reduce dependencies. If it's difficult to build an object then that's a symptom of a bigger problem, complex models with too many responsibilities.

Reply:
that's an interesting point. Mine is more about the fact that sample data needs to be deterministic IMO. We all (not only testers) need to always know the data that gets set in the system so that we can test without having to learn it each time from scratch.

If I know `Enterprise X` has a members-only shop and I'm asked to test a feature of those, I can go straight ahead and do my tests with it.

Reply:
What I think of FacoryBot here is: it works, it makes our life easer. It's not production code. No problem :-)

Reply:
If it was production code I'd recommend not using factorybot and move the building code into a "Builder pattern" object that supports object creation.

Reply:
I don't care so much about using FactoryBot but more about the reproducibility (whathever word :troll:) of our sample data.

Reply:
I am not sure I understand exactly what change you are requesting.
Is it to replace factorybot with something else? If that is the case, if you agree, we'd approve this PR and create a separate issue to tackle that.

Reply:
we need a reproducible state. Every time we run sample data we need to get the same result, so we better create the shipping method without FactoryBot as the latter will always generate random data, I believe. What do you think?

Reply:
yes, I understand that and agree. But I dont think this is specific to this change or to shipping_method. If you look at this file. It's using FactoryBot for most of the data. So, we would need to rewrite this file and verify it creates the same data all the time.

Reply:
Oh, you are right. I didn't realise that it was using FactoryBot already. Sorry. Different scope. :-)

Reply:
ok, then I give up :+1: